docs/specs/code/EXPORT_TRANSACTION_HISTORY_TO_FILE.md

```diff
@@ -0,0 +1,114 @@
+
+This library might be checked for example: https://libraries.io/npm/xlsx
+
+## Exported file format
+
+[Transaction_Export_Example.xlsx](https://github.com/Emurgo/yoroi-frontend/files/2700970/Transaction_Export_Example.xlsx)
+
+![image](https://user-images.githubusercontent.com/5585355/50316566-42ab3e80-04c8-11e9-8f38-e45343472545.png)
+
+We want to support this exact file format to be compatible with other services.
+
+Columns:
+1. **Type** - for now only `Deposit` or `Withdrawal`
+2. **Buy Amount** - non-empty only for `Deposit`
+3. **Buy Cur.** - non-empty only for `Deposit`, and for now only always `ADA`
+4. **Sell Amount** - non-empty only for `Withdrawal`
+5. **Sell Cur.** - non-empty only for `Withdrawal`, and for now only always `ADA`
+6. **Fee Amount** - non-empty only for `Withdrawal` (because for deposit fee is not included)
+7. **Fee Cur.** - non-empty only for `Withdrawal`, and for now only always `ADA`
+8. **Exchange** - for now always empty (user might enter manually)
+9. **Trade Group** - for now always empty
+10. **Comment** - for now always empty
+11. **Date** - transaction time
+
+When file is produced - it should be named:
+```
+Yoroi-Transaction-History_YYYY-MM-DD.{xlsx, csv}
+```
+
+With **current date** inserted.
+
+# Iteration 2 (export period)
+
+## Abstract
+
+Now users should be able to chose the time period to export transaction for, like day from and day to. There also might be a list of default periods, e.g.: "this month", "last 30 days", "this quarter", "this year", "all". Default period selected might be "this month".
+
+## UI
+
+Export dialog now contains additional elements to select a date-period:
+
+![image](https://user-images.githubusercontent.com/5585355/50401307-9eaee500-079e-11e9-8972-a8162a3a5982.png)
+
+Dialog contains:
+1. Comment about the fact that the whole existing history will be exported
+2. Dropdown-selector: "File format"
+3. Dropdown-selector "Period"
+4. Two date selectors: "from day", "to day"
+5. Button "Export"
+6. Icon "Help" (Question mark)
+
+"Period" selector contains multiple **dynamic** values (meaning that selecting the same value
+in different calendar days will select different specific period), like: "this month", "last month",
+"this quarter", "last quarter", "this year", "last year", etc. While any of these **dynamic** values
+are selected - "From" and "To" fields change their value accordingly, but they remain **disabled**
+for manual input.
+
+Additionally "Period" selector contains special value **"Custom"**. When use selects this option -
+"From" and "To" selectors become enabled for manual input. 
+
+## Technical
+
+The difference now is when transactions are selected from the local storage - 
+start and end dates are inserted as filters.
+
+## File naming
+
+When file is produced - it should be named:
+```
+Yoroi-Transaction-History_YYYY-MM-DD_to_YYYY-MM-DD.{xlsx, csv}
```

Comment:
For Iteration-1 I am thinking to use file name as:
`Yoroi-Transaction-History.{xlsx, csv}`

Reply:
For iteration 1 it might be only 1 date - when document is generated (time is not necessary, imo), so if I were to export a file today - `Yoroi-Transaction-History_2018-01-11`

Reply:
Makes sense,
will use: `Yoroi-Transaction-History_YYYY-MM-DD.{xlsx, csv}` for Iteration1